vendors/espressif/boards/esp32/ports/wifi/iot_wifi.c

```diff
@@ -263,24 +263,48 @@ BaseType_t WIFI_IsConnected( void )
 
 WIFIReturnCode_t WIFI_Off( void )
 {
+    esp_err_t ret;
+    if ((ret = esp_wifi_deinit()) != ESP_OK) {
+        if (ret == ESP_ERR_WIFI_NOT_STOPPED) {
+            ret = esp_wifi_disconnect();
+            if (ret != ESP_OK) {
+                ESP_LOGE(TAG, "%s:Failed to disconnect wifi %d", __func__, ret);
+                goto err;
+            }
+            ret = esp_wifi_stop();
+            if (ret != ESP_OK) {
+                ESP_LOGE(TAG, "%s:Failed to stop wifi %d", __func__, ret);
+                goto err;
+            }
+            if (esp_wifi_deinit() != ESP_OK) {
+                ESP_LOGE(TAG, "%s:Failed to deinit %d", __func__, ret);
+                goto err;
+            }
```

Comment:
delete `wifi_event_group` otherwise it can cause memory leak when WIFI_On() is called.